pymc/logprob/mixture.py

```diff
@@ -433,6 +408,53 @@ def logprob_MixtureRV(
     return logp_val
 
 
+class MeasurableSwitchMixture(MeasurableElemwise):
+    valid_scalar_types = (Switch,)
+
+
+MeasurableVariable.register(MeasurableSwitchMixture)
+
+
+measurable_switch_mixture = MeasurableSwitchMixture(scalar_switch)
+
+
+@node_rewriter([switch])
+def find_measurable_switch_mixture(fgraph, node):
+    rv_map_feature: Optional[PreserveRVMappings] = getattr(fgraph, "preserve_rv_mappings", None)
+
+    if rv_map_feature is None:
+        return None  # pragma: no cover
+
+    switch_cond, *components = node.inputs
+
+    # We don't support broadcasting of components (as those values are not independent).
```

Comment:
Yes. `switch(idx, normal(), normal(size=(5,))` broadcast (copies) the entries of the scalar branch in the forward graph. So the correct logp would need to not overcount those / return 0 when they are different (since that's not possible). Because the logp does not do that, we fail the rewrite. I'll make it more explicit in the comment

Reply:
Expanded comment